src/pip/_internal/distributions/__init__.py

```diff
@@ -0,0 +1,23 @@
+from pip._internal.distributions.source import SourceDistribution
+from pip._internal.distributions.wheel import WheelDistribution
+
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from pip._internal.distributions.base import AbstractDistribution
+    from pip._internal.req.req_install import InstallRequirement
+
+
+def make_abstract_dist(install_req):
```

Comment:
I would call this simply `make_distribution()` rather than `make_abstract_dist()` since it returns an instance of one of the concrete `Distribution` subclasses and not an object whose type equals `AbstractDistribution`. The "abstract" in `AbstractDistribution` is a reference to the fact that it's an abstract class and so shouldn't be instantiated directly. (I know one can argue that the return value is an instance of `AbstractDistribution, but I think it's better to avoid the confusion so people looking at the calling code or type annotation aren't led to believe that the return value has type equal to `AbstractDistribution`.)

Reply:
Sound right to me; I'll do this in a follow up where I cleanup the names here. :)